compute/zgbset.c

```diff
@@ -49,14 +59,16 @@ void plasma_zgbset(int m, int n, int kl, int ku,
         for(; extralower > 0; extralower--)
         {
             cornerI = m-extralower;
-            plasma_zlaset(PlasmaGeneral,1,extralower,0,0,pA+cornerI,lda+1);
+            plasma_zlaset(PlasmaGeneral,1,(extralower < n ? extralower: n),
+                                          0,0,pA+cornerI,lda+1);
         }
         for(; extraupper > 0; extraupper--)
         {
             cornerJ = n-extraupper;
             plasma_zlaset(PlasmaGeneral,1,extraupper,0,0,pA+lda*cornerJ,lda+1);
         }
         if(m>n) // m strictly greater
+                // (clears a "tail" on bottom right below diag)
```

Comment:
Consider a matrix `a` which is `10` by `5`, with `kl=kl=2`. The routine will call zlaset first to clear out the majority of the matrix properly. Namely, it will clear everything except for some elements in the bottom left corner, and three additional elements need zero but are zeroed yet. Namely, these elements are (indexing as [column][row]) `a[4][7],a[4][8],a[4][9]`.

I agree that calling these a "tail" is less than ideal, but I wanted to at least put something down as a comment. I'll try to improve the comments here as well.